adds blank line ignore

lattice0 opened the pull request:
ignores blank line for cases where sdp has a blank line and adds test example from a real camera that has this 'problem'

sdroege commented on src/parser.rs:
```diff
@@ -734,15 +738,14 @@ impl<'iter, 'item, I: Iterator<Item = (usize, &'item [u8])>> FallibleIterator
                 })??;
 
             if key == self.current {
```
You could also do

```suggestion
            return if key == self.current {
```

And add a `;` at the final `}`. But works for me either way :)

sdroege commented on src/parser.rs:
```diff
@@ -715,7 +715,11 @@ impl<'iter, 'item, I: Iterator<Item = (usize, &'item [u8])>> FallibleIterator
     fn next(&mut self) -> Result<Option<Self::Item>, Self::Error> {
-        if let Some((n, line)) = self.it.peek() {
+        while let Some((n, line)) = self.it.peek() {
+            if line.len() == 0 {
```
```suggestion
            if line.is_empty() {
```

sdroege reviewed (commented):
Looks good to me otherwise :) Can you squash all commits into a single one with a meaningful commit message and `git push --force` them into this PR?

Also feel free to add a second commit with the `Cargo.toml` changes to reduce the number of dependencies from `bstr`.

sdroege commented on src/parser.rs:
```diff
@@ -737,12 +737,12 @@ impl<'iter, 'item, I: Iterator<Item = (usize, &'item [u8])>> FallibleIterator
             } else {
-                return Err(ParserError::UnexpectedLine(*n, key))
+                Err(ParserError::UnexpectedLine(*n, key))
             }
```
```suggestion
            };
```

lattice0 commented:
Did you like the new commit?

sdroege commented:
> Did you like the new commit?

Yes, apart from the missing `;` and the missing `cargo fmt` run this is good to be merged now :)

lattice0 committed:
Ignore blank lines in the SDP

While the RFC does not specify handling of blank lines, various software inserts them and skipping over them would make it possible to parse their SDPs.

lattice0 commented:
I think it's ok now

sdroege reviewed (commented):
Looks good to me (I'll fix the clippy warnings) but can you write a more descriptive commit message? Something like

```
Ignore blank lines in the SDP

While the RFC does not specify handling of blank lines, various software inserts them and skipping over them would make it possible to parse their SDPs.
```

sdroege committed:
Fix various clippy warnings

sdroege committed:
Fix some more clippy warnings

lattice0 commented:
What do you think of it now?

sdroege commented:
Thanks, looks all good now!
